src/components/timeline/Ruler.js

```diff
@@ -57,13 +74,14 @@ export class TimelineRuler extends PureComponent<Props> {
 
   _getNotches() {
     if (this.props.width === 0) {
-      return { notches: [], notchTime: 0 };
+      return { notches: [], notchTime: 0, unit: '' };
     }
 
-    const { zeroAt, rangeStart, rangeEnd, width } = this.props;
+    const { zeroAt, rangeStart, rangeEnd, width, unit } = this.props;
     const pixelsPerMilliSecond = width / (rangeEnd - rangeStart);
     const minimumNotchWidth = 55; // pixels
     const notchTime = this._findNiceNumberGreaterOrEqualTo(
+      unit,
       minimumNotchWidth / pixelsPerMilliSecond
```

Comment:
optional: Might be clearer to move `minimumNotchWidth / pixelsPerMilliSecond` into a separate variable called `minMillisecondsPerNotch`.